shared/services/config/networks-config.go

```diff
@@ -0,0 +1,71 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/rocket-pool/smartnode/shared/types/config"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	defaultNetworksConfigPath = "networks-default.yml"
+	extraNetworksConfigPath   = "networks-extra.yml"
+)
+
+type NetworksConfig struct {
```

Comment:
seems like a great opportunity to use `type NetworksConfig []*config.NetworkInfo`.

You'll have to remove the `networks:` piece from the smartnode-install .yml file and just have the list be top-level.

but then, instead of:

`defaultNetworks.Networks = append(defaultNetworks.Networks, extraNetworks.Networks...)` below you can ergonmically `return append(defaultNetworks, extraNetworks...)`

and you can kill off `func (nc *NetworksConfig) GetNetworks() []*config.NetworkInfo `

Reply:
My thought here was future proofing: We may want other information about that config file rather than just an array of NetworkInfo at some point. 

I don't feel strongly though. Let me know if you prefer though the flat array and I can change it.

Reply:
Good point, it's easier to extend if the root level is an object. I kind of am leaning towards:
```golang
type NetworksConfigs []*config.NetworkInfo
type NetworksConfigFile struct {
	Networks []NetworkConfigs `yaml:"networks"`
}
```
now... that way we get the extensibility of the file format _and_ can simplify the ABI here to remove GetNetworks().

If we ever have to add more fields in memory and change the `NetworkConfigs` type down the line, the compiler will yell at us about the surface area, so it's any easy upgrade path.

Reply:
Seems like a good option as well. I think this one is ready for @0xfornax but i'll give it one last pass